juicer/workflow/workflow.py

```diff
@@ -1,27 +1,263 @@
+                                    task.get('operation')['id'])
+
+            workflow_graph.add_node(
+                task.get('id'),
+                tahiti_indegree_required=query_tahiti['N_INPUT'],
+                tahiti_indegree_multiplicity=query_tahiti['M_INPUT'],
+                tahiti_outdegree_required=query_tahiti['N_OUTPUT'],
+                tahiti_outdegree_multiplicity=query_tahiti['M_OUTPUT'],
+                attr_dict=task)
+
+        for flow in flows:
+            workflow_graph.add_edge(flow['source_id'], flow['target_id'],
+                                         attr_dict=flow)
+
+        print self.workflow_graph.nodes()
+        # updating in_degree and out_degree
+        for nodes in self.workflow_graph.nodes():
+            self.workflow_graph.node[nodes]['in_degree'] = self.workflow_graph. \
+                in_degree(nodes)
+            self.workflow_graph.node[nodes]['out_degree'] = self.workflow_graph. \
+                out_degree(nodes)
+        print self.workflow_graph.nodes()
+        return workflow_graph
+
+
+    def plot_workflow_graph_image(self):
+        """
+             Show the image from workflow_graph
+        """
+        # Change layout according to necessity
+        pos = nx.spring_layout(self.workflow_graph)
+        nx.draw(self.workflow_graph, pos, node_color='#004a7b', node_size=2000,
+                edge_color='#555555', width=1.5, edge_cmap=None,
+                with_labels=True, style='dashed',
+                label_pos=50.3, alpha=1, arrows=True, node_shape='s',
+                font_size=8,
+                font_color='#FFFFFF')
+        plt.show()
+        # If necessary save the image
+        # plt.savefig(filename, dpi=300, orientation='landscape', format=None,
+        # bbox_inches=None, pad_inches=0.1)
+
+    def get_topological_sorted_tasks(self):
+
+        """ Create the tasks Graph and perform topological sorting
+
+            A topological sort is a nonunique permutation of the nodes
+            such that an edge from u to v implies that u appears before
+            v in the topological sort order.
+
+            :return: Return a list of nodes in topological sort order.
+        """
+        # First, map the tasks IDs to their original position
+        tasks_position = {}
+
+        for count_position, task in enumerate(self.workflow_data['tasks']):
+            tasks_position[task['id']] = count_position
+
+        sorted_tasks_id = nx.topological_sort(self.workflow_graph, reverse=False)
+
+        return sorted_tasks_id
+
+    def check_null_source_id_tasks(self):
+        for flow in self.workflow_data['flows']:
+            if flow['source_id'] == "":
+                return 0
+        return 1
+
+    def check_null_target_id_tasks(self):
+        for flow in self.workflow_data['flows']:
+            if flow['target_id'] == "":
+                return 0
+        return 1
+
+
+    def get_reversed_graph(self):
```

Comment:
Remove this function. It contains only 1 statement and do not need to be separated.